062-UseKRaft-feature-gate-promotion.md

```diff
@@ -0,0 +1,95 @@
+    * Scale-up of dedicated controller nodes (the scale-up eventually succeeds, but the broker availability is not maintained)
+  * This is currently blocked by the support for [KIP-853](https://cwiki.apache.org/confluence/display/KAFKA/KIP-853%3A+KRaft+Controller+Membership+Changes).
+    This KIP might not be implemented earlier than in Apache Kafka 4.0.
+    Support for scaling controllers is currently not seen as blocker for moving the `UseKRaft` feature gate to _GA_.
+* JBOD support
+  * Tracked in [strimzi/strimzi-kafka-operator#9437](https://github.com/strimzi/strimzi-kafka-operator/issues/9437)
+  * Expected to be implemented in Apache Kafka 3.7.0.
+    The Strimzi implementation fill follow the release of Kafka 3.7.0 in Strimzi 0.40 or 0.41.
+
+## Proposed timeline
+
+This proposal proposes the following timeline for the graduation of the `UseKRaft` feature gate:
+* Move to _beta_ phase and be enabled by default in Strimzi 0.40.0
+* Move to _GA_ phase and be permanently enabled in Strimzi 0.42.0
+
+It is worth noting that:
+* In addition to the `UseKRaft` feature gate, the KRaft clusters are enabled / disabled using the `strimzi.io/kraft` annotation.
+  As a result, moving the `UseKRaft` feature gate to _beta_ or _GA_ does not mean that all new Kafka clusters have to use KRaft or that this has any impact on existing ZooKeeper based clusters.
+  KRaft will be applied only to the Kafka clusters with the right annotation.
+* Moving the `UseKRaft` feature to beta or GA does not mean we will drop support for ZooKeeper-based clusters.
+  While moving the `UseKRaft` feature gate to _GA_ defines the earliest moment when support for ZooKeeper based clusters can be dropped, it is currently not expected to happen right after Strimzi 0.42 and this proposal does not establish any such plan.
+
+Moving the `UseKRaft` feature gate to _beta_ or _GA_ serves mainly the following purpose:
+* Indicate the progress in Kraft implementation and improved production-readiness.
+* Make it easier to run KRaft clusters without the need to enable any feature gate.
+
+## Proposed changes
+
+Apart from the change to the feature gate status itself, this proposal also includes several other changes.
+
+### CRD changes
+
+KRaft is usable only with the use of `KafkaNodePool` custom resources that configure the number of replicas and storage configuration.
+As part of promoting the `UseKRaft` feature gate to _beta_ stage, we will make the following fields in the `Kafka` custom resource optional instead of required:
+* `.spec.kafka.replicas`
+* `.spec.kafka.storage`
+
+In addition, the KRaft clusters do not need the ZooKeeper configuration.
+So the `.spec.zookeeper` section will be made optional as well.
+
+For ZooKeeper clusters, the validation of these fields will be done inside the Strimzi operator.
+Additionally, CEL validation rules will be considered as they might allow us to do additional validation on the Kubernetes level.
```

Comment:
Should we say a bit more about how the validation might be broken down by operator and Kubernetes level?